rustls/src/server/tls12.rs

```diff
@@ -403,17 +403,19 @@ impl hs::State for ExpectFinished {
             .map(|_| verify::FinishedMessageVerified::assertion())?;
 
         // Save session, perhaps
-        if !self.resuming && !self.handshake.session_id.is_empty() {
-            let value = get_server_session_value_tls12(&self.secrets, self.using_ems, sess);
-
-            let worked = sess.config.session_storage.put(
-                self.handshake.session_id.get_encoding(),
-                value.get_encoding(),
-            );
-            if worked {
-                debug!("Session saved");
-            } else {
-                debug!("Session not saved");
+        if !self.resuming {
+            if let Some(session_id) = self.handshake.session_id {
```

Comment:
To prevent the increased indentation here, maybe try `if let (Some(session_id), false) = (self.handshake.session_id, self.resuming)`?